src/entity.ts

```diff
@@ -52,27 +52,44 @@ export namespace entity {
    * provided.
    *
    * @class
-   * @param {number} value The double value.
+   * @param {number|string} value The double value.
    *
    * @example
    * const {Datastore} = require('@google-cloud/datastore');
    * const datastore = new Datastore();
    * const aDouble = datastore.double(7.3);
    */
-  export class Double {
+  export class Double extends Number {
+    private _entityPropertyName: string | undefined;
     type: string;
     value: number;
-    constructor(value: number) {
+    constructor(value: number | string | ValueProto) {
+      super(typeof value === 'object' ? value.doubleValue : value);
+
       /**
        * @name Double#type
        * @type {string}
        */
       this.type = 'DatastoreDouble';
+
+      this._entityPropertyName =
+        typeof value === 'object' ? value.propertyName : undefined;
+
       /**
        * @name Double#value
        * @type {number}
        */
-      this.value = value;
+      this.value =
+        typeof value === 'object' ? Number(value.doubleValue) : Number(value);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    valueOf(): any {
+      return Number(this.value);
+    }
+
+    toJSON(): any {
+      return {type: this.type, value: this.value};
```

Comment:
This seems quite dangerous if the goal of `Double` is to fake a `Number` ?

Reply:
can you expand? Sorry, I don't quite understand/follow. Also, `class Int` has a similar toJson.

Reply:
I am really not sure to get the intent of the change right but my understanding of the current state is that you want to return `Int`s and `Double`s that users would use as if they were `Number`s.

If I get this right then `Int`s and `Double`s should behave like `Number`s as much a possible.

However with this code:
- `JSON.stringify(1.23) == "1.23"`
- but `JSON.stringify(Double(1.23)) == '{"type": ..., "value": ...}'`

While not related to `toJSON()`  returning wrapped numbers (`Int` / `Double`) could be problematic:

```
const [entity] = await ds.get(...);

// entity.price is a Double, i.e. Double(10)

entity.price *= 0.90;

// entity.price is now a number (9, "int")

// warning + DB screwed
await ds.save(entity);
```